programs/intent-transfer/src/lib.rs

```diff
@@ -28,6 +36,368 @@ pub mod intent_transfer {
+    pub ntt_peer: UncheckedAccount<'info>,
+
+    /// CHECK: check not important per https://github.com/wormholelabs-xyz/example-ntt-with-executor-svm/blob/10c51da84ee5deb9dee7b2afa69382ce90984eae/programs/example-ntt-with-executor-svm/src/lib.rs#L78-L80
+    #[account(mut)]
+    pub ntt_outbox_item: Signer<'info>,
+
+    /// CHECK: check not important per https://github.com/wormhole-foundation/native-token-transfers/blob/8bd672c5164c53d5a3f9403dc7ce3450da539450/solana/programs/example-native-token-transfers/src/queue/outbox.rs#L50
+    #[account(mut)]
+    pub ntt_outbox_rate_limit: UncheckedAccount<'info>,
+
+    /// CHECK: checked in NTT manager program
+    #[account(mut)]
+    pub ntt_custody: UncheckedAccount<'info>,
+
+    /// CHECK: checked in NTT with executor program
+    #[account(mut)]
+    pub payee_ntt_with_executor: UncheckedAccount<'info>,
+}
+
+#[derive(Accounts)]
+pub struct BridgeNttTokens<'info> {
+    #[account(seeds = [chain_id::SEED], seeds::program = chain_id::ID, bump)]
+    pub from_chain_id: Account<'info, ChainId>,
+
+    /// CHECK: we check the address of this account
+    #[account(address = instructions::ID)]
+    pub sysvar_instructions: UncheckedAccount<'info>,
+
+    /// CHECK: this is just a signer for token program CPIs
+    #[account(seeds = [INTENT_TRANSFER_SEED], bump)]
+    pub intent_transfer_setter: UncheckedAccount<'info>,
+
+    pub token_program: Program<'info, Token>,
+
+    #[account(mut, token::mint = mint)]
+    pub source: Account<'info, TokenAccount>,
+
+    #[account(
+        init_if_needed,
+        payer = sponsor,
+        seeds = [BRIDGE_NTT_INTERMEDIATE_SEED, source.key().as_ref()],
+        bump,
+        token::mint = mint,
+        token::authority = intent_transfer_setter
+    )]
+    pub intermediate_token_account: Account<'info, TokenAccount>,
+
+    #[account(mut)]
+    pub mint: Account<'info, Mint>,
+
+    pub metadata: Option<UncheckedAccount<'info>>,
+
+    #[account(
+        init_if_needed,
+        payer = sponsor,
+        space = Nonce::DISCRIMINATOR.len() + Nonce::INIT_SPACE,
+        seeds = [BRIDGE_NTT_NONCE_SEED, source.owner.key().as_ref()],
+        bump
+    )]
+    pub nonce: Account<'info, Nonce>,
+
+    #[account(mut)]
+    pub sponsor: Signer<'info>,
+
+    /// This signer exists solely to protect against MITM attacks whereby a malicious actor
+    /// intercepts a submitted transaction prior to any signatures being attached and changes
+    /// fields not directly encoded by the intent message. For example, such an attack could
+    /// alter the relay instructions in the bridging; attaching this signature prior to the
+    /// transaction leaving the client protects against that attack vector. Neither this signer
+    /// nor its signature are substantively required for this transaction. Note we do not need
+    /// to perform any checks on this signer; its mere presence is sufficient.
+    pub session_signer: Signer<'info>,
```

Comment:
this signer is unchecked, I don't think it prevents the MITM attack: I can still intercept the intent, replace this signer by a key I control and sign the new transaction with that key